app/controllers/grant_submissions/submissions/reviews_controller.rb

```diff
@@ -66,13 +66,16 @@ def update
         authorize @review
         respond_to do |format|
           if @review.update(review_params)
-            set_redirect_path
-            format.html { redirect_to @redirect_path,
-                          notice: 'Review was successfully updated.' }
+            format.js { render json: { success: true } }
+            format.html do
+              set_redirect_path
+              redirect_to @redirect_path,
+                          notice: 'Review was successfully updated.'
+            end
             format.json { render :show, status: :ok, location: @review }
           else
+            format.js   { render json: { errors: @review.errors.full_messages, success: false } }
             flash.now[:alert] = @review.errors.full_messages
-            build_criteria_reviews
```

Comment:
should this have been removed?

Reply:
I added it back, but build_criteria_reviews already runs when entering the edit page in the first place, so it seemed duplicated.